x-pack/plugins/ml/public/embeddables/types.ts

```diff
@@ -50,16 +59,17 @@ export interface AnomalySwimlaneEmbeddableUserInput {
 }
 
 export interface AnomalySwimlaneEmbeddableCustomInput {
+  id?: string;
   jobIds: JobId[];
   swimlaneType: SwimlaneType;
   viewBy?: string;
   perPage?: number;
 
   // Embeddable inputs which are not included in the default interface
-  filters: Filter[];
-  query: Query;
-  refreshConfig: RefreshInterval;
-  timeRange: TimeRange;
+  filters?: Filter[];
```

Comment:
nit: I'd either rename this type or merge it into `AnomalySwimLaneEmbeddableState`.

Additionally, do you need `filters`, `query`, and `refreshConfig` in here? Those should be provided by the parent API.

You can get the time range type from `SerializedTimeRange` also.

Reply:
@ThomThomson I just checked the usage and other plugins use this type to initialize the anomaly swim lane embeddable, e.g. observability [here](https://github.com/elastic/kibana/blob/2f9b90a9ea5a3f3dffecb9caec05db2066870b00/x-pack/plugins/observability_solution/infra/public/pages/logs/log_entry_rate/sections/anomalies/anomalies_swimlane_visualisation.tsx#L44). Good you pointed that out, because we need to update the embeddable renderer component!